### PR TITLE
feat: support slog attributes

### LINKED ISSUE #119: improve json handling of `slog.Group` fields
was checking this out as a way of improving displayed logs however noticed that `slog.Group` fields are represented differently for both the text and json formatter

Here is the output from the sample code included below
```
INFO charm text formatter req="[method=POST url=example.com]" map=map[foo:bar] list="[1 2 3]"
level=INFO msg="slog text formatter" req.method=POST req.url=example.com map=map[foo:bar] list="[1 2 3]"
{"level":"info","list":"[1 2 3]","map":"map[foo:bar]","msg":"charm json formatter","req":"[method=POST url=example.com]"}
{"level":"INFO","msg":"slog json formatter","req":{"method":"POST","url":"example.com"},"map":{"foo":"bar"},"list":[1,2,3]}
```

for the text formatter while `req.method & req.url` would be preferred it is still usable. however for the json output the representation is not a nested map, this becomes less valuable since parsing that data becomes more difficult as cannot use the same json parsing/selecting expressions with `jq`
ex. to get the method i could use `jq '.req.method'` however that will no longer work with the charm json formatter

Update: seems to not be specific to `group` using `map or list` value also does not get encoded as json

Here is the sample code for reproduction 
```go
package main

import (
	"log/slog"
	"os"

	"github.com/charmbracelet/log"
)

func RemoveTime(groups []string, a slog.Attr) slog.Attr {
	if a.Key == slog.TimeKey && len(groups) == 0 {
		return slog.Attr{}
	}
	return a
}

func main() {
	ct := slog.New(log.NewWithOptions(os.Stdout, log.Options{}))
	cj := slog.New(log.NewWithOptions(os.Stdout, log.Options{Formatter: log.JSONFormatter}))
	st := slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{ReplaceAttr: RemoveTime}))
	sj := slog.New(slog.NewJSONHandler(os.Stdout, &slog.HandlerOptions{ReplaceAttr: RemoveTime}))

	g := slog.Group("req",
		slog.String("method", "POST"),
		slog.String("url", "example.com"),
	)

	m := map[string]string{"foo": "bar"}
	l := []int{1, 2, 3}

	ct.Info("charm text formatter", g, "map", m, "list", l)
	st.Info("slog text formatter", g, "map", m, "list", l)
	cj.Info("charm json formatter", g, "map", m, "list", l)
	sj.Info("slog json formatter", g, "map", m, "list", l)
}
```

### PR DESCRIPTION
**Pull request by op**
This adds support for slog attributes and fixes #119.

### CODE REVIEW

**Comment by op** [comment]
@aymanbagabas very cool idea! that worked pretty nice.

File: slog_no121.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	slogKindBool
+	slogKindDuration
+	slogKindFloat64
+	slogKindInt64
+	slogKindString
+	slogKindTime
+	slogKindUint64
+	slogKindGroup
+	slogKindLogValuer
+)
```
Comment (aymanbagabas) [inline slog_no121.go:32]: Here, we could simply use `golang.org/x/exp/slog`

File: logger_121_test.go [hunk 2]
```diff
@@ -183,3 +182,89 @@ func TestSlogCustomLevel(t *testing.T) {
...
+		},
+		{
+			name:     "string",
+			expected: `{"level":"info","msg":"message","string":"hello"}` + "\n",
+			kvs:      []any{"string", slog.StringValue("hello")},
+		},
+		{
+			name:     "time",
+			expected: `{"level":"info","msg":"message","_time":"1970-01-01T00:00:00Z"}` + "\n",
+			kvs:      []any{"_time", slog.TimeValue(time.Unix(0, 0).UTC())},
```
[the remark this answers is not in the document]
Comment (aymanbagabas) [reply logger_121_test.go:228]: Why does it fail?

**Review by aymanbagabas — COMMENTED** [review]
Some nit picks and questions, looks good otherwise!

File: slog_no121.go (hunk 1 above)
Comment (op) [reply slog_no121.go]: Great suggestion! :+1:

File: logger_121_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (op) [reply logger_121_test.go:228]: I updated the PR so it's super simple to change the behavior: just move the `default` case and add `return` for all ok cases. I'm unsure how these fields are supposed to be handled though and behavior for both text and json should be the same.

**Comment by op** [comment]
@aymanbagabas hi, did you see that all comments has been addressed? :)

**Comment by aymanbagabas** [comment]
Hey @op! Thanks for addressing the comments. I tried running the example from #119, but I think the implementation still doesn't support arbitrary maps, which is the whole point of the issue.

**Comment by op** [comment]
@aymanbagabas ping! :penguin:

File: logger_121.go [hunk 3]
```diff
@@ -27,7 +36,7 @@ func (l *Logger) Handle(ctx context.Context, record slog.Record) error {
 
 	fields := make([]interface{}, 0, record.NumAttrs()*2)
 	record.Attrs(func(a slog.Attr) bool {
-		fields = append(fields, a.Key, a.Value.String())
+		fields = append(fields, a.Key, a.Value)
```
Comment (aymanbagabas) [inline logger_121.go:39]: Probably want to do the same for `logger_no121.go`

**Review by aymanbagabas — COMMENTED** [review]
LGTM, some last comments before merging

File: logger_121.go (hunk 3 above)
Comment (op) [reply logger_121.go:39]: Fixed!

**Comment by op** [comment]
@aymanbagabas seems like govulncheck doesn't work with 1.19. Looks like https://github.com/golang/vuln/commit/745db65f35062023c5994fceb1931ef839777007 introduces `slices`.

**Comment by aymanbagabas** [comment]
> @aymanbagabas seems like govulncheck doesn't work with 1.19. Looks like [golang/vuln@745db65](https://github.com/golang/vuln/commit/745db65f35062023c5994fceb1931ef839777007) introduces `slices`.

Hmm, seems like golang/vuln needs to update their go.mod

Anyway, this looks good, thanks again @op!

**Pull request merged**